client/src/components/profile/components/HeatMap.js

```diff
@@ -87,27 +128,124 @@ function HeatMap({ calendar, streak }) {
               'data-tip': `<b>${valueCount}</b> ${dateFormatted}`
             };
           }}
-          values={calendarData}
+          values={dataToDisplay}
         />
         <ReactTooltip className='react-tooltip' effect='solid' html={true} />
+
+        <Spacer />
+        <Row>
+          <div className='streak-container'>
+            <span className='streak'>
+              <b>Longest Streak:</b> {longestStreak || 0}
+            </span>
+            <span className='streak'>
+              <b>Current Streak:</b> {currentStreak || 0}
+            </span>
+          </div>
+        </Row>
+        <hr />
       </FullWidthRow>
-      <Spacer />
-      <FullWidthRow>
-        <div className='streak-container'>
-          <span className='streak'>
-            <b>Longest Streak:</b> {streak.longest || 0}
-          </span>
-          <span className='streak'>
-            <b>Current Streak:</b> {streak.current || 0}
-          </span>
-        </div>
-      </FullWidthRow>
-      <Spacer />
-      <hr />
-    </FullWidthRow>
-  );
+    );
+  }
 }
 
+HeatMapInner.propTypes = innerPropTypes;
+
+const HeatMap = props => {
+  const { calendar } = props;
+
+  /**
+   *  the following logic creates the data for the heatmap
+   *  from the users calendar and calculates their streaks
+   */
+
+  // create array of timestamps and turn into milliseconds
+  const timestamps = Object.keys(calendar).map(stamp => stamp * 1000);
+  const startOfTimestamps = startOfDay(new Date(timestamps[0]));
+  let endOfCalendar = startOfDay(Date.now());
+  let startOfCalendar;
+
+  // creates pages for heatmap
+  let pages = [];
+
+  do {
+    startOfCalendar = addDays(addMonths(endOfCalendar, -6), 1);
+
+    const newPage = {
+      startOfCalendar: startOfCalendar,
+      endOfCalendar: endOfCalendar
+    };
+
+    pages.unshift(newPage);
```

Comment:
```suggestion
    pages.push(newPage);
```
Isn't `unshift` a bit slow in JS? How about `push` and `reverse`?

Reply:
Sounds good to me.